internal/ethapi/transaction_args.go

```diff
@@ -177,6 +177,14 @@ func (args *TransactionArgs) setDefaults(ctx context.Context, b Backend) error {
 
 // setFeeDefaults fills in default fee values for unspecified tx fields.
 func (args *TransactionArgs) setFeeDefaults(ctx context.Context, b Backend) error {
+	head := b.CurrentHeader()
+	// Sanity check the EIP-4844 fee parameters.
+	if args.BlobFeeCap != nil && args.BlobFeeCap.ToInt().Sign() == 0 {
+		return errors.New("maxFeePerBlobGas must be non-zero")
+	}
+	if err := args.setCancunFeeDefaults(ctx, head, b); err != nil {
```

Comment:
previously, `setCancunFeeDefaults` also internally called `setLondonFeeDefaults`, which makes sense to me -- makes it so it by default recursively does The Right Thing. You changed it so we now must make two explicit invocations. 
Why?

Reply:
I decoupled them. Reason is sometimes we should set cancun fields but we can skip london fields, as they are already provided by the user.

Reply:
I think it makes sense because `setCancunFeeDefaults` is now just one step in a series of steps.